core/lib/libtomcrypt/src/mpa_desc.c

```diff
@@ -489,7 +489,9 @@ static int montgomery_reduce(void *a, void *b, void *c)
 	LTC_ARGCHK(b != NULL);
 	LTC_ARGCHK(c != NULL);
 	mpanum tmp;
-	init((void **)&tmp);
+	if (init((void **)&tmp) != CRYPT_OK) {
+		return CRYPT_MEM;
```

Comment:
Well, I do not think it saves memory... I would say that in case the function is not inlined, returning another error may require the CRYPT_MEM assignement to return.

But well, this is not the issue.
Let's keep your way so. We can make further cleanup if necessary.